File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n == 0 || n == 1)
        {
            return 0;
        }
        int res = 0;
        vector<bool> sieve(n, true);
        sieve[0] = false;
        sieve[1] = false;
        for(int i = 2; i <= sqrt(n); i++)
        {
            if(sieve[i])
            {
                for(int j = i + i; j < n; j += i)
                {
                    sieve[j] = false;
                }
            }
        }
        for(int i = 0; i < n; i++)
        {
            if(sieve[i])
            {
                res++;
            }
        }
        return res;
    }
};

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n == 0 or n == 1:
            return 0
        sieve = [True for _ in range(n)]
        sieve[0] = False
        sieve[1] = False
        for i in range(2, int(sqrt(n)) + 1):
            if sieve[i]:
                for j in range(i + i, n, i):
                    sieve[j] = False
        return sieve.count(True)
